### ISSUE TITLE
Method name encoding seems off

### ISSUE THREAD

**Issue by headius** [jruby/jruby#2127]
This case from mri/ruby/test_syntax:

```ruby
  def test_error_message_encoding
    bug10114 = '[ruby-core:64228] [Bug #10114]'
    code = "# -*- coding: utf-8 -*-\n" "def n \"\u{2208}\"; end"
    assert_syntax_error(code, /def n "\u{2208}"; end/, bug10114)
  end
```

fails with:

```
TestSyntax#test_error_message_encoding [/Users/headius/projects/jruby/test/mri/ruby/test_syntax.rb:473]:
[ruby-core:64228] [Bug #10114].
Expected /def n "\u{2208}"; end/ to match "[ruby-core:64228] [Bug #10114]:2: syntax error, unexpected tSTRING_BEG\ndef n \"â\u0088\u0088\"; end\n      ^".
```

Unsure where the fix is needed but I'll mark parser and core.

**Comment by jodosha** [comment]
Hello @headius and @enebo I've found an error with encoding method name, but not sure if it's related to this. See https://github.com/lotus/utils/commit/e38490099113c4e6e1ad5878050ca826f7fabe3c

For this reason I reported the problem here, instead of opening a new issue. Cheers :)

**Comment by enebo** [comment]
@jodosha this is not related but if you could try out nightlies I think your original source should work now (on Jruby 9000): http://ci.jruby.org/.  Let me know if not..

**Comment by jodosha** [comment]
@enebo The problem is still there:

```shell
➜ ruby -v
jruby 9.1.0.0-SNAPSHOT (2.3.0) 2016-04-01 30c1276 Java HotSpot(TM) 64-Bit Server VM 25.60-b23 on 1.8.0_60-b27 +jit [darwin-x86_64]

(master) ~/Code/hanami/utils:jruby-9.1.0.0-SNAPSHOT ➜ g revert --no-commit e38490099113c4e6e1ad5878050ca826f7fabe3c
*(master) ~/Code/hanami/utils:jruby-9.1.0.0-SNAPSHOT ➜ ber
NameError: Undefined method test_0001_escapes 'тест' for class '#<Class:0x10027fc9>'
                                                                         undef_method at org/jruby/RubyModule.java:2710
                                                                          block in it at /Users/luca/.gem/jruby/2.3.0/gems/minitest-5.8.4/lib/minitest/spec.rb:225
                                                                                 each at org/jruby/RubyArray.java:1593
                                                                                   it at /Users/luca/.gem/jruby/2.3.0/gems/minitest-5.8.4/lib/minitest/spec.rb:224
                                                                      block in (root) at /Users/luca/Code/hanami/utils/test/escape_test.rb:105
                                                                          module_eval at org/jruby/RubyModule.java:2719
                                                                             describe at /Users/luca/.gem/jruby/2.3.0/gems/minitest-5.8.4/lib/minitest/spec.rb:83
                                                                      block in (root) at /Users/luca/Code/hanami/utils/test/escape_test.rb:8
                                                                          module_eval at org/jruby/RubyModule.java:2719
                                                                             describe at /Users/luca/.gem/jruby/2.3.0/gems/minitest-5.8.4/lib/minitest/spec.rb:83
                                                                                <top> at /Users/luca/Code/hanami/utils/test/escape_test.rb:5
                                                                              require at org/jruby/RubyKernel.java:944
  block in /Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb at /Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb:10
                                                                                 each at org/jruby/RubyArray.java:1593
  block in /Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb at /Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb:9
                                                                               select at org/jruby/RubyArray.java:2411
                                                                                <top> at /Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb:4
rake aborted!
Command failed with status (1): [ruby -w -I"lib:test" -I"/Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib" "/Users/luca/.gem/jruby/2.3.0/gems/rake-11.1.1/lib/rake/rake_test_loader.rb" "test/**/*_test.rb" ]

Tasks: TOP => default => test
(See full trace by running task with --trace)
```

**Comment by enebo** [comment]
@jodosha can you open up a new issue for this?  It will be a little gory to untangle since it appears rspec will make a method and then once it is finished it tries to undef it (does not know how rspec is implemented)?  It looks like it executes but then when it is time to remove it we are passing a different set of bytes/String to find the method definition.  At least it is a guess.

**Comment by jodosha** [comment]
@enebo Sure, there it is: https://github.com/jruby/jruby/issues/3778

**Comment by enebo** [comment]
@jodosha yeah look's perfect!  Thanks.
